packages/@aws-cdk/aws-cognito/lib/user-pool-domain.ts

```diff
@@ -126,4 +130,32 @@ export class UserPoolDomain extends Resource implements IUserPoolDomain {
     });
     return customResource.getResponseField('DomainDescription.CloudFrontDistribution');
   }
+
+  /**
+   * The URL to the hosted UI associated with this domain
+   */
+  public baseUrl(): string {
+    if (this.isCognitoDomain) {
+      return `https://${this.domainName}.auth.${Stack.of(this).region}.amazoncognito.com`;
+    }
+    return `https://${this.domainName}`;
+  }
+
+  /**
+   * The URL to the sign in page in this domain using a specific UserPoolClient
+   * @param client [disable-awslint:ref-via-interface] the user pool client that the UI will use to interact with the UserPool
+   * @param redirectUri the URI to redirect to after sign in. This URI must be one specified in the `callbackUrl`
```

Comment:
Not too crucial but thought i'd throw it out there:

How about we expose `callbackUrls` on `UserPoolClient` so that we can validate the `redirectUrl` argument here?

Reply:
I'll leave that for a future PR. Maybe a community contribution that someone would want to do ;)